app/services/assets/handlers/java.js

```diff
@@ -0,0 +1,163 @@
+    ];
+
+    for (let index = 0; index < jdbcMatches.length; index++) {
+      const match = jdbcMatches[index];
+      const path = match[1].trim();
+      inputs.push({
+        id: `JDBC - ${path}`,
+        type: Constants.DependencyType.DATA,
+        path,
+      });
+    }
+
+    return inputs;
+  }
+
+  getOutputs(uri, text) {
+    const outputs = [];
+    if (!text || text.trim() === '') {
+      return outputs;
+    }
+
+    // For file write operations
+    const fileWriteMatches = [
+      ...text.matchAll(/new\s+(FileOutputStream|FileWriter|BufferedWriter|PrintWriter)\s*\(\s*(?:new\s+File\s*\(\s*)?(['"]{1,}\s*?[\s\S]+?['"]{1,})[\s\S]*?\)/gim),
+      ...text.matchAll(/Files\.(?:write|writeString|newBufferedWriter|newOutputStream)\s*\(\s*(?:Paths\.get\s*\(\s*)?(['"]{1,}\s*?[\s\S]+?['"]{1,})[\s\S]*?\)/gim),
+    ];
+
+    for (let index = 0; index < fileWriteMatches.length; index++) {
+      const match = fileWriteMatches[index];
+      const operation = match[1] || 'Files.write';
+      const path = match.length > 2 ? match[2].trim() : match[1].trim();
+      outputs.push({
+        id: `${operation} - ${path}`,
+        type: Constants.DependencyType.DATA,
+        path,
+      });
+    }
+
+    // For image write operations
+    const imageWriteMatches = [
+      ...text.matchAll(/ImageIO\.write\s*\(\s*[\s\S]*?,\s*['"]{1,}[\s\S]+?['"]{1,}\s*,\s*(?:new\s+File\s*\(\s*)?(['"]{1,}\s*?[\s\S]+?['"]{1,})[\s\S]*?\)/gim),
+    ];
+
+    for (let index = 0; index < imageWriteMatches.length; index++) {
+      const match = imageWriteMatches[index];
+      const path = match[1].trim();
+      outputs.push({
+        id: `ImageIO.write - ${path}`,
+        type: Constants.DependencyType.FIGURE,
+        path,
+      });
+    }
+
+    // For chart export operations
+    const chartExportMatches = [
+      ...text.matchAll(/ChartUtilities\.saveChartAs(?:JPEG|PNG)\s*\(\s*[\s\S]*?,\s*(?:new\s+File\s*\(\s*)?(['"]{1,}\s*?[\s\S]+?['"]{1,})[\s\S]*?\)/gim),
+      ...text.matchAll(/ChartUtils\.saveChartAs(?:JPEG|PNG)\s*\(\s*[\s\S]*?,\s*(?:new\s+File\s*\(\s*)?(['"]{1,}\s*?[\s\S]+?['"]{1,})[\s\S]*?\)/gim),
+    ];
+
+    for (let index = 0; index < chartExportMatches.length; index++) {
+      const match = chartExportMatches[index];
+      const path = match[1].trim();
+      outputs.push({
+        id: `Chart Export - ${path}`,
+        type: Constants.DependencyType.FIGURE,
+        path,
+      });
+    }
+
+    return outputs;
+  }
+
+  getLibraries(uri, text) {
+    const libraries = [];
+    if (!text || text.trim() === '') {
+      return libraries;
+    }
+
+    const packageMatches = [
+      ...text.matchAll(/package\s+([\w.]+)\s*;/gm),
+    ];
+
+    let currentPackage = '';
+    if (packageMatches.length > 0) {
+      currentPackage = packageMatches[0][1];
+    }
+
+    const importMatches = [
+      ...text.matchAll(/import\s+(?:static\s+)?([\w.]+)(?:\.(\*|\w+))?\s*;/gm),
```

Comment:
Is this working as designed?  It looks like you're trying to explicitly capture `*` as part of the last capture group (which is good!), but when I tested the regex specifically it wasn't detecting that.  I can see on Line 151 that you are handling if that group is missing, but it'd be nice if the regex was working as intended.